store/file_store.go

```diff
@@ -216,3 +231,32 @@ func itemSubdir(key string) string {
 	}
 	return result
 }
+
+// Some Simple Item Key Validations
+func isKeyValid(key string) (bool, error) {
```

Comment:
I wonder if this function could be changed to just return `error`. Then a valid key would have a `nil` returned, and an invalid key would have the appropriate error returned. If you do this (and please do!) add something to the comment about returning nil means the key is valid.